junto-rust/zomes/core/code/src/expressions/definitions/function_definitions.rs

```diff
@@ -0,0 +1,200 @@
+use hdk::{
+    holochain_core_types::{
+        cas::content::Address, 
+        hash::HashString,
+        json::{
+            JsonString,
+            default_to_json
+        },
+        error::HolochainError,
+        dna::capabilities::CapabilityRequest
+    }
+};
+
+use std::collections::HashMap;
+use serde::Serialize;
+
+use super::app_definitions;
+
+#[derive(Serialize, Deserialize, Debug, DefaultJson)]
+pub struct Env {
+    pub dna_name: String,
+    pub dna_address: String,
+    pub agent_id: String,
+    pub agent_address: String,
+    pub cap_request: CapabilityRequest,
+}
+
+#[derive(Serialize, Deserialize, Debug, PartialEq, Clone)]
+pub struct CreateUserInformation{
+    pub username: String,
+    pub first_name: String,
+    pub last_name: String,
+    pub profile_picture: String,
+    pub bio: String
+}
+
+//Basic struct to be used to describe a function and its parameters to the handle_hooks & handle_contextual_links functions
+pub struct FunctionDescriptor{  
+    pub name: &'static str,
+    pub parameters: FunctionParameters,
+}
+
+#[derive(Serialize, Deserialize)]
+pub struct UserDens{
+    pub private_den: Option<GetLinksLoadElement<app_definitions::Channel>>,
+    pub shared_den: Option<GetLinksLoadElement<app_definitions::Channel>>,
+    pub public_den: Option<GetLinksLoadElement<app_definitions::Channel>>
+}
+
+#[derive(Serialize, Deserialize)]
+pub struct UserPack{
+    pub pack: Option<GetLinksLoadElement<app_definitions::Group>>
+}
+
+#[derive(Debug, Serialize, Deserialize)]
+pub struct GroupMembers{
+    pub members: Vec<GetLinksLoadElement<app_definitions::UserName>>
+}
+
+#[derive(Debug, Serialize, Deserialize)]
+pub struct ExpressionResults<T>{
+    pub expressions: Vec<GetLinksLoadElement<T>>
+}
+
+#[derive(Debug, Serialize, Deserialize)]
+pub enum QueryTarget{
+    ExpressionPost,
+    User
+}
+
+#[derive(Debug, Serialize, Deserialize)]
+pub enum QueryOptions {
+    FilterPopular,
+    FilterNew,
+    FilterOld
+}
+
+#[derive(Debug, Serialize, Deserialize)]
+pub enum QueryType {
+    And,
+    Or
+}
+
+pub type GetLinksLoadResult<T> = Vec<GetLinksLoadElement<T>>;
+
+#[derive(Serialize, Deserialize, Debug, Clone)]
```

Comment:
Can you not just derive `DefaultJson` and put some trait bounds on T (T: DefaultJson). You shouldn't have to implement the conversion yourself. If that doesn't work we might need to make some changes to `DefaultJson`